staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -443,6 +443,55 @@ func NewEventCorrelator(clock clock.Clock) *EventCorrelator {
 	}
 }
 
+func NewEventCorrelatorWithOptions(options *CorrelatorOptions) *EventCorrelator {
```

Comment:
nit, use `CorrelatorOptions` instead of `*CorrelatorOptions` so we get our own copy and avoid mutating what is passed in

Reply:
Done.

Also, thanks for the quick review :)